Chess/desk.h

```diff
@@ -1,25 +1,36 @@
 #ifndef DESK_H
 #define DESK_H
 
-#include <QObject>
+#include "common_types.h"
+#include "figure.h"
+
 #include <vector>
 
+
 class Desk : public QObject
 {
     Q_OBJECT
 public:
+    friend class Command;

```

Comment:
avoid using friend relationship! it's the strongest relationship can ever exist.
create a separate interface to pass it to command object